libcontainer/factory_linux.go

```diff
@@ -261,19 +265,26 @@ func (l *LinuxFactory) StartInitialization() (err error) {
 			fmt.Fprintln(os.Stderr, err)
 			return
 		}
-		// ensure that this pipe is always closed
-		pipe.Close()
 	}()
 	defer func() {
 		if e := recover(); e != nil {
 			err = fmt.Errorf("panic from initialization: %v, %v", e, string(debug.Stack()))
 		}
 	}()
+
+	var i initer
 	i, err = newContainerInit(it, pipe, rootfd)
 	if err != nil {
 		return err
 	}
-	return i.Init()
+
+	if i, err = newContainerInit(it, pipe, rootfd); err != nil {
+		return
```

Comment:
We don't use this form of `return`. Please use `return err` _explicitly_, as well as `i, err :=`.